src/core/plugin/PluginController.cpp

```diff
@@ -1,54 +1,137 @@
+    if (!fs::is_directory(path)) {
+        g_info("Skipping Plugin path, it is not a directory: %s", path.string().c_str());
+        return {};
+    }
+    g_info("Loading plugins from: %s", path.string().c_str());
 
+    std::vector<std::unique_ptr<Plugin>> returner;
     try {
         for (auto const& f: fs::directory_iterator(path)) {
             const auto& pluginPath = f.path();
-            auto plugin = std::make_unique<Plugin>(control, pluginPath.filename().string(), pluginPath);
-            if (!plugin->isValid()) {
-                g_warning("Error loading plugin \"%s\"", f.path().string().c_str());
-                continue;
+            try {
+                auto plugin = std::make_unique<Plugin>(control, pluginPath.filename().string(), pluginPath);
+                if (!plugin || !plugin->isValid()) {
+                    g_warning("Error loading plugin \"%s\"", f.path().string().c_str());
+                    continue;
+                }
+                plugin->setEnabled(plugin->isDefaultEnabled());
+                returner.emplace_back(std::move(plugin));
+            } catch (std::exception const& e) {
+                g_warning("Error loading plugin \"%s\": %s", f.path().string().c_str(), e.what());
             }
+        }
+    } catch (fs::filesystem_error const& e) {
+        g_warning("Could not open plugin dir: \"%s\": %s", path.string().c_str(), e.what());
+    }
+    return returner;
+}
 
-            if (plugin->isDefaultEnabled()) {
-                plugin->setEnabled(!(std::find(pluginDisabled.begin(), pluginDisabled.end(), plugin->getName()) !=
-                                     pluginDisabled.end()));
-            } else {
-                plugin->setEnabled(std::find(pluginEnabled.begin(), pluginEnabled.end(), plugin->getName()) !=
-                                   pluginEnabled.end());
-            }
+auto emplace_sorted_if_not_exists(std::vector<std::unique_ptr<Plugin>>& plugins, std::unique_ptr<Plugin> plugin)
+        -> void {
+    assert(std::is_sorted(begin(plugins), end(plugins), PluginComparator{}));
+    auto iter = std::equal_range(begin(plugins), std::end(plugins), plugin, PluginComparator{});
+    if (iter.first != iter.second) {
+        return;
+    }
+    plugins.emplace(iter.first, std::move(plugin));
+    assert(std::is_sorted(begin(plugins), end(plugins), PluginComparator{}));
+}
+}  // namespace
+#endif
 
-            plugin->loadScript();
 
-            this->plugins.emplace_back(std::move(plugin));
+PluginController::PluginController(Control* control): control(control) {
+#ifdef ENABLE_PLUGINS
+    // Todo(fabian) move those search paths into PathUtils
+    auto searchPath = control->getGladeSearchPath()->getFirstSearchPath();
+    auto searchPaths = {fs::weakly_canonical(searchPath /= "../plugins"),  //
+                        fs::weakly_canonical(searchPath /= "/plugins"),    //
```

Comment:
The `searchPath` is the `ui` subfolder of the [Shared resources folder](https://xournalpp.github.io/guide/file-locations/#shared-resources-folder), at least if it exists. If the `ui`-subfolder does not exist, it would be the shared resources folder itself, but I don't think Xournal++ can run without the `ui`-subfolder.
So I don't understand why adding `searchPath /= "/plugins"` is needed.

Reply:
Fixed